WordPressKit/BlazeServiceRemote.swift

```diff
@@ -12,20 +12,18 @@ open class BlazeServiceRemote: ServiceRemoteWordPressComREST {
     open func searchCampaigns(forSiteId siteId: Int, page: Int = 1, callback: @escaping (Result<BlazeCampaignsSearchResponse, Error>) -> Void) {
         let endpoint = "sites/\(siteId)/wordads/dsp/api/v1/search/campaigns/site/\(siteId)"
         let path = self.path(forEndpoint: endpoint, withVersion: ._2_0)
-        wordPressComRestApi.GET(path, parameters: [
-            "page": "\(page)" as AnyObject
-        ], success: { response, _ in
-            do {
-                let data = try JSONSerialization.data(withJSONObject: response)
-                let response = try JSONDecoder.apiDecoder.decode(BlazeCampaignsSearchResponse.self, from: data)
-                callback(.success((response)))
-            } catch {
-                WPKitLogError("Error parsing campaigns response: \(error), \(response)")
-                callback(.failure(error))
-            }
-        }, failure: { error, _ in
-            WPKitLogError("Error retrieving blaze campaigns: ", error)
-            callback(.failure(error))
-        })
+        Task { @MainActor in
+            let result = await self.wordPressComRestApi
+                .perform(
+                    .get,
+                    URLString: path,
+                    parameters: ["page": page] as [String: AnyObject],
+                    jsonDecoder: JSONDecoder.apiDecoder,
+                    type: BlazeCampaignsSearchResponse.self
+                )
+                .map { $0.body }
+                .eraseToError()
+            callback(result)
```

Comment:
Thinking out loud. Given we'll be using this pattern a bunch, we could add a syntax sugar to write it like

```swift
            await wordPressComRestApi
                .perform(
                    .get,
                    URLString: path,
                    parameters: ["page": page] as [String: AnyObject],
                    jsonDecoder: JSONDecoder.apiDecoder,
                    type: BlazeCampaignsSearchResponse.self
                )
                .map { $0.body }
                .eraseToError()
                .forward(to: callback)
```

Reply:
👍 I'll make the change in a upcoming PR.